### PR TITLE
lndclient: add SubscribeChannelEvents

### PR DESCRIPTION
**Pull request by bhandras**

### CODE REVIEW

File: lightning_client.go [hunk 1]
```diff
@@ -1463,6 +1511,155 @@ func (s *lightningClient) ChannelBackups(ctx context.Context) ([]byte, error) {
...
+
+	result := &ChannelEventUpdate{}
+	update := rpcChannelEventUpdate.Channel
+	var err error
+
+	switch rpcChannelEventUpdate.Type {
+	case lnrpc.ChannelEventUpdate_PENDING_OPEN_CHANNEL:
+		result.UpdateType = PendingOpenChannelUpdate
+
+		channelPoint := update.(*lnrpc.ChannelEventUpdate_PendingOpenChannel).PendingOpenChannel
```
Comment (guggero) [inline lightning_client.go:1541]: nit: can use gRPC helper methods like `rpcChannelEventUpdate.GetPendingOpenChannel()` for this and below.

**Review by guggero — APPROVED** [review]
Very clean and straightforward, LGTM :100:

File: lightning_client.go [hunk 2]
```diff
@@ -1453,6 +1511,155 @@ func (s *lightningClient) ChannelBackups(ctx context.Context) ([]byte, error) {
...
+			fundingTxId.FundingTxidBytes,
+			channelPoint.OutputIndex,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+	case lnrpc.ChannelEventUpdate_INACTIVE_CHANNEL:
+		result.UpdateType = InactiveChannelUpdate
+		channelPoint := update.(*lnrpc.ChannelEventUpdate_ActiveChannel).ActiveChannel
```
Comment (guggero) [inline lightning_client.go:1599]: Copy/paste error, must be inactive channel!

**Review by guggero — CHANGES_REQUESTED** [review]
Found a bug while testing, see comment above.

File: lightning_client.go (hunk 2 above)
Comment (bhandras) [reply lightning_client.go]: Thanks, indeed that's a copy paste bug, fixed.

File: lightning_client.go (hunk 1 above)
Comment (bhandras) [reply lightning_client.go]: Yeah, that's much nicer, done.

**Review by guggero — APPROVED** [review]

**Pull request merged**